src/styles/_calendar.scss

```diff
@@ -0,0 +1,55 @@
+@use 'sass:list';
+@use 'variables';
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: variables.$gap;
+  padding: 10px;
+  width: calc((#{variables.$day-size} * #{variables.$columns}) + (#{variables.$gap} * #{variables.$columns - 1}));
+  background-color: variables.$calendar-background-color;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    border: 1px solid variables.$border-color;
+    width: variables.$day-size;
+    height: variables.$day-size;
+    background-color: variables.$day-background-color;
+
+    &::before {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      margin: auto;
+    }
+    &:hover {
+      cursor: pointer;
+      background-color: variables.$hover-background-color;
+      transform: translateY(-20px);
+      transition:
+        background-color 0.5s,
+        transform 0.5s;
+    }
+  }
+
+  @each $day in variables.$days {
+    $index: list.index(variables.$days, $day) - 1;
+    &--start-day-#{$day} .calendar__day:nth-child(1) {
+      margin-left: $index * (variables.$day-size + 1);
```

Comment:
There is a potential issue with the calculation of the margin-left property. The multiplication should consider the gap between days as well, so it should be $index * (variables.$day-size + variables.$gap).